.github/workflows/maven.yml

```diff
@@ -28,3 +28,40 @@ jobs:
       run: mvn -B package --file pom.xml --settings $GITHUB_WORKSPACE/settings.xml
       env:
         GITHUB_TOKEN: ${{ github.token }}
+
+    - name: Generate JaCoCo Badges for core/api
+      id: coreApi
+      uses: cicirello/jacoco-badge-generator@v2
+      with:
+        generate-branches-badge: true
+        jacoco-csv-file: core/api/target/site/jacoco/jacoco.csv
+        coverage-badge-filename: jacoco1.svg
+        branches-badge-filename: branches1.svg
+
+    - name: Generate JaCoCo Badges for core/parser
+      id: coreParser
+      uses: cicirello/jacoco-badge-generator@v2
+      with:
+        generate-branches-badge: true
+        jacoco-csv-file: core/parser/target/site/jacoco/jacoco.csv
+        coverage-badge-filename: jacoco2.svg
+        branches-badge-filename: branches2.svg
+
+    - name: Generate JaCoCo Badges for core/vertx
+      id: coreVertx
+      uses: cicirello/jacoco-badge-generator@v2
+      with:
+        generate-branches-badge: true
+        jacoco-csv-file: core/vertx/target/site/jacoco/jacoco.csv
+        coverage-badge-filename: jacoco3.svg
+        branches-badge-filename: branches3.svg
+
+    - name: Commit the badge (if it changed)
+      run: |
+        if [[ `git status --porcelain` ]]; then
+          git config --global user.name 'Thomas Pliakas'
```

Comment:
We probably need to use an account that's not your personal one here :-).  Is there another way?

Reply:
GITHUB_TOKEN, perhaps?

Reply:
Agree with @dsgrieve !